docs/guides/universal-receiver-delegate/create-custom-urd-2.md

```diff
@@ -0,0 +1,294 @@
+
+- grants permission to the URD to call the UP (SUPER_CALL + REENTRANT) - Method 1
+- calls `authorizeOperator()` on the LSP7 token to authorize the URD to spend token on the UP's behalf - Method 2
+
+Create a file called `deployLSP1.ts` in your `scripts/` folder with the following content:
+
+```ts title="scripts/deployLSP1.ts"
+import hre from 'hardhat';
+import { ethers } from 'hardhat';
+import * as dotenv from 'dotenv';
+import {
+  ERC725YDataKeys,
+  LSP1_TYPE_IDS,
+  PERMISSIONS,
+} from '@lukso/lsp-smart-contracts';
+
+// load env vars
+dotenv.config();
+
+// You can update the value of the allowed LSP7 token
+const contractsAddr = ['0x63890ea231c6e966142288d805b9f9de7e0e5927'];
+
+// Update those values in the .env file
+const { UP_ADDR, PRIVATE_KEY, UP_RECEIVER, PERCENTAGE } = process.env;
+
+/**
+ * In this script, we will:
+ * - deploy our URD implementation (LSP1URDForwarder.sol)
+ * - setDataBatch on the UP to register URD implementation + permission for URD contract
+ */
+async function main() {
+  // ----------
+  // BASE SETUP
+  // ----------
+
+  // setup provider
+  const provider = new ethers.JsonRpcProvider(
+    'https://rpc.testnet.lukso.network',
+  );
+  // setup signer (the browser extension controller)
+  const signer = new ethers.Wallet(PRIVATE_KEY as string, provider);
+  // load the associated UP
+  const UP = await ethers.getContractAt('UniversalProfile', UP_ADDR as string);
+  console.log('🔑 EOA: ', signer.address);
+  console.log('🆙 Universal Profile: ', await UP.getAddress());
+
+  // ----------
+  // DEPLOY URD
+  // ----------
+
+  console.log('⏳ Deploying the custom URD');
+  const CustomURDBytecode =
+    hre.artifacts.readArtifactSync('LSP1URDForwarder').bytecode;
+
+  // we need to encode the constructor parameters and add them to the contract bytecode
+  const abiCoder = new ethers.AbiCoder();
+  const params = abiCoder
+    .encode(
+      ['address', 'uint256', 'address[]'],
+      [UP_RECEIVER as string, PERCENTAGE as string, contractsAddr],
+    )
+    .slice(2);
+  const fullBytecode = CustomURDBytecode + params;
+
+  // get the address of the contract that will be created
+  const CustomURDAddress = await UP.connect(signer).execute.staticCall(
+    1,
+    ethers.ZeroAddress,
+    0,
+    fullBytecode,
+  );
+
+  // deploy LSP1URDForwarder as the UP (signed by the browser extension controller)
+  const tx1 = await UP.connect(signer).execute(
+    1,
+    ethers.ZeroAddress,
+    0,
+    fullBytecode,
+  );
+  await tx1.wait();
+
+  console.log(
+    '✅ Custom URD successfully deployed at address: ',
+    CustomURDAddress,
+  );
+
+  // -------------------------
+  // GRANT URD PERM - METHOD 1
+  // -------------------------
+  // COMMENT IF YOU USE METHOD 2
+
+  // we need the key to store our custom URD contract address
+  // {_LSP1_UNIVERSAL_RECEIVER_DELEGATE_PREFIX + <bytes32 typeId>}
+  console.log('⏳ Registering custom URD on the UP');
+  const URDdataKey =
```

Comment:
Would that be easier to use _erc725.js_ here? Or would that overcomplicate the guide?

Reply:
maybe for another time :D